core/opendaq/functionblock/include/opendaq/function_block_impl.h

```diff
@@ -272,6 +279,27 @@ void FunctionBlockImpl<TInterface, Interfaces...>::updateObject(const Serialized
                      { updateInputPort(localId, obj, context); });
     }
 
+    if (obj.hasKey("FB"))
+    {
+        const auto fbFolder = obj.readSerializedObject("FB");
+        fbFolder.checkObjectType("Folder");
+        auto serializedItems = this->getSerializedItems(fbFolder);
+
+        if (!serializedItems.empty())
+        {
+            auto availableFbs = onGetAvailableFunctionBlockTypes();
+            for (const auto& [id, serializedItem] : serializedItems)
+            {
+                serializedItem.checkObjectType("FunctionBlock");
+                const auto typeId = serializedItem.readString("typeId");
+                if (!availableFbs.hasKey(typeId))
+                {
+                    throw NotSupportedException("Function block type not supported");
+                }
```

Comment:
Not sure that I'd throw an exception here, I'd maybe log an error. Otherwise we crash the whole update.